### PR TITLE
feat: remove MUB size threshold

### PR DESCRIPTION
**Pull request by pauldix**
Removes the MUB chunk close based on size. Also add a check in lifecycle policy to move if the MUB chunk crosses a default row count threshold.

### CODE REVIEW

**Review by alamb — APPROVED** [review]
I think this code does what it says, though I am not sure what the rationale is for this change

Can we also please (if not in this PR then we can file a ticket for a follow on):
1. Remove `mutable_size_threshold` from the protobuf rules / `DbRules` struct
2. track making the value configurable

File: server/src/db.rs [hunk 1]
```diff
@@ -687,8 +683,7 @@ impl Db {
                                 continue;
                             }
 
-                            let new_chunk = partition.create_open_chunk(mb_chunk);
-                            check_chunk_closed(&mut *new_chunk.write(), mutable_size_threshold);
+                            let _ = partition.create_open_chunk(mb_chunk);
```
Comment (tustvold) [inline server/src/db.rs:686]:
```suggestion
                            partition.create_open_chunk(mb_chunk);
```

File: server/src/db.rs [hunk 2]
```diff
@@ -1821,40 +1803,6 @@ mod tests {
         assert!(chunk.time_closed().unwrap() < after_rollover);
     }
 
-    #[tokio::test]
```
Comment (tustvold) [inline server/src/db.rs:1824]: It would be nice to have a test of the logic introduced by this change

**Review by tustvold — APPROVED** [review]
I think a test of some form would be warranted - I'm also somewhat surprised this hasn't broken any of the end-to-end tests....

**Comment by pauldix** [comment]
@alamb added #1878 to track reworking the MUB config rules. Also added a few other tracking issues before you commented (in activity feed).

File: server/src/db.rs (hunk 2 above)
Comment (alamb) [reply server/src/db.rs:1824]: 100% agree

**Comment by pauldix** [comment]
@tustvold added a test in 44df03306a05719f9a245a1c55845d6f857ffeda

File: server/src/db.rs (hunk 2 above)
Comment (pauldix) [reply server/src/db.rs:1824]: We don't close anymore so this test isn't valid. This happens through the lifecycle policy by passing one of the threshold. I added a test for moving based on that in 61917c107f3ef0f6b7db8aa6dfae8fb409069d15. I've also logged #1877 to track using a row count threshold to close a MUB chunk here in `db.rs`

File: server/src/db.rs [hunk 3]
```diff
@@ -2443,7 +2391,7 @@ mod tests {
                 ("access", "exclusive"),
             ])
             .counter()
-            .eq(2.)
+            .eq(1.)
```
Comment (pauldix) [inline server/src/db.rs:2394]: @tustvold This failed after removal of the `can_close` function. I assume it was due to some lock not happening and thus not incrementing this counter. Can you verify this is the case? It wasn't clear to me what this part of the test was for.

Comment (tustvold) [reply server/src/db.rs:2394]: Yup, that sounds correct to me :+1:

**Review by tustvold — APPROVED** [review]

**Pull request merged**